hera_pspec/grouping.py

```diff
@@ -416,32 +472,54 @@ def average_spectra(uvp_in, blpair_groups=None, time_avg=False,
                 bpg_nsmp = np.sum(bpg_nsmp, axis=0)
                 bpg_ints = np.sum(bpg_ints, axis=0) \
                          / np.sum(w_list, axis=0).clip(1e-10, np.inf)
+                if store_cov:
+                    bpg_cov = np.sum(bpg_cov, axis=0) \
+                            / np.sum(w_list, axis=0)[:,None].clip(1e-10, np.inf)
                 w_list = np.sum(w_list, axis=0)
-                
+
+                # For errors that aren't 0, sum weights and take inverse square
+                # root. Otherwise, set to invalid value -99.
+                for stat in stat_l:
+                    arr = np.sum(bpg_stats[stat], axis=0)
+                    not_valid = np.isclose(arr, 0., 1e-10)
```

Comment:
I still think this should be
```
not_valid = np.isclose(arr, 0, 1e-10) + arr < 0
```
because the error estimation may have done something weird and assigned a non-zero but negative error and that will screw up the averaging downstream...